torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -0,0 +1,194 @@
+#include <torch/csrc/jit/codegen/cuda/arith.h>
+#include <torch/csrc/jit/codegen/cuda/ir_iostream.h>
+#include <torch/csrc/jit/codegen/cuda/lower_utils.h>
+
+#include <torch/csrc/jit/codegen/cuda/lower_thread_predicate.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+const static std::unordered_map<ParallelType, int> pt_to_offset{
+    {ParallelType::BIDx, 0},
+    {ParallelType::BIDy, 1},
+    {ParallelType::BIDz, 2},
+    {ParallelType::TIDx, 3},
+    {ParallelType::TIDy, 4},
+    {ParallelType::TIDz, 5}};
+
+const static std::unordered_map<int, ParallelType> offset_to_pt{
+    {0, ParallelType::BIDx},
+    {1, ParallelType::BIDy},
+    {2, ParallelType::BIDz},
+    {3, ParallelType::TIDx},
+    {4, ParallelType::TIDy},
+    {5, ParallelType::TIDz}};
+
+static constexpr int num_p_type = 6;
+
+namespace {
```

Comment:
nit: I'd recommend moving away from anonymous namespaces for helper functions (they tend to work against natural helper locality, and the debugging/profiling experience suffers)